engine/classes/Elgg/ActionRequest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+namespace Elgg;
+
+/**
+ * Models the API handed to an action handler
+ *
+ * @access private
+ */
+class ActionRequest implements \Elgg\Services\ActionRequest {
+
+	private $name;
+	private $elgg;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Application $elgg Elgg application
+	 * @param string      $name Action name
+	 */
+	public function __construct(Application $elgg, $name) {
+		$this->elgg = $elgg;
```

Comment:
I'm torn about exposing a bunch more services. I don't like that it would yield multiple ways of doing most everything, but it's the only way to add functionality to existing unexposed services.